content/en/docs/guides/metadata.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Metadata
+description: >-
+  Explains what metadata is, how it is transmitted, and what it is used for.
+---
+
+### Overview
+
+Metadata is a side channel that allows clients and servers to provide
+information to each other that is associated with the data messages.
+
+gRPC metadata is a key-value pair of data that is sent with a gRPC request or
+response. It is used to provide additional information about the call, such as
+authentication credentials, tracing information, or custom headers.
+
+gRPC metadata is implemented using HTTP/2 headers. The keys and values are ASCII
+strings, but the values can also be binary data. The keys are case insensitive
+and must not start with the prefix grpc-, which is reserved for gRPC itself.
+
+gRPC metadata can be sent and received by both the client and the server. It is
+sent in the initial headers of the RPC call, and can also be sent in subsequent
+trailers.
+
+gRPC metadata is useful for a variety of purposes, such as:
+
+* **Authentication**: gRPC metadata can be used to send authentication credentials to
+the server. This can be used to implement different authentication schemes, such
+as OAuth2 or JWT.
+* **Tracing**: gRPC metadata can be used to send tracing information to the server.
+This can be used to track the progress of a request through a distributed
+system.
+* **Custom headers**: gRPC metadata can be used to send custom headers to the server.
+This can be used to implement application-specific features, such as load
+balancing, rate limiting or providing detailed error messages from the server
+to the client.
+* **Internal usages**: metadata is used by gRPC to communicate how to handle an
+RPC such as compression, keep-alive and other things.  It is also used to
+communicate the Status of an RPC from the server to the client.
+
+See [Concepts guide](https://grpc.io/docs/guides/concepts.html#metadata)
+
+#### Be Aware
+```
+WARNING: Servers may limit the size of Request-Headers, with a default of 8 KiB suggested.
+```
+
+Custom metadata must follow the "Custom-Metadata" format listed in
+[PROTOCOL-HTTP2](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md)
+, with the exception of binary headers not needing to be base64 encoded.
+
+
+
+#### Headers
+Headers are sent before any data.  The header includes things like
```

Comment:
"before any data" makes it sound like this can't be used on server-side.

Reply:
reworded